src/services/helpers.js

```diff
@@ -82,3 +82,83 @@ export function countFractionDigits(number = 0) {
 export function isHalf(number) {
   return Math.abs(number) % 1 === 0.5
 }
+
+/**
+ * Fetches a JSON resource.
+ * @ignore
+ *
+ * @param  {String} url - The resource to fetch.
+ * @param  {Object} options.headers - The headers to pass.
+ *
+ * @throws {Error} If `request.status` is lesser than 200 or greater or equal to 400.
+ * @throws {Error} If network fails.
+ *
+ * @return {JSON}
+ */
+export function getJSON(url, options = {}) {
+  return new Promise((resolve, reject) => {
+    const request = Object.assign(new XMLHttpRequest(), {
+      onreadystatechange() {
+        if (request.readyState === 4) {
+          if (request.status >= 200 && request.status < 400)
+            resolve(JSON.parse(request.responseText))
+          else reject(new Error(request.statusText))
+        }
+      },
+      onerror() {
+        reject(new Error('Network error'))
+      }
+    })
+
+    request.open('GET', url, true)
+    setXHRHeaders(request, options.headers)
+    request.send()
+  })
+}
```

Comment:
Not sure if there is an explicit browser-support contract, but any reason not to use `fetch` here? Seems like any browser that supports `Object#assign` would also support `fetch`.

Reply:
I initially used it but when it was an absolute pain when it came to testing it with Jest, and I couldn't find a way to do it without yet another dev dependency. I may be wrong though, would be happy to know if there's a simpler solution.

Reply:
That's what I thought, `Fetch` needs [Babel Polyfill](https://babeljs.io/docs/usage/polyfill/) as a **dependency** (and not a dev dependency) to be made backward-compatible with IE11. Since this is a library and not an app, I'm a bit reluctant to adding dependencies unless this is absolutely necessary.

Reply:
I'd like to be backward-compatible at least down to IE11 (I shipped polyfilled versions today in [v1.2.0](https://github.com/sarahdayan/dinero.js/releases/tag/v1.2.0) as I realized there were some holes). I picked [core-js](https://github.com/zloirock/core-js) for polyfills (the same that Babel uses), and unfortunately, it doesn't provide one for `Fetch`.

I want to avoid adding dependencies for each feature, so let's wait for this one and refactor later when it's time to drop IE altogether.